Userland/Libraries/LibCore/Directory.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Format.h>
+#include <AK/LexicalPath.h>
+#include <AK/Noncopyable.h>
+#include <AK/Optional.h>
+#include <LibCore/Stream.h>
+#include <dirent.h>
+#include <sys/stat.h>
+
+namespace Core {
+
+class DirIterator;
+
+// Deal with real system directories. Any Directory instance always refers to a valid existing directory.
+class Directory {
+    AK_MAKE_NONCOPYABLE(Directory);
+
+public:
+    Directory(Directory&&) = default;
+
+    // When this flag is set, both the directory attempted to instantiate as well as all of its parents are created with mode 0755 if necessary.
+    enum class CreateDirectories : bool {
```

Comment:
the `CreateDirectories` is a bit confusing. `CreateParents` would make more sense

Reply:
No, because it also creates the specified directory if that doesn't exist.